pvlib/irradiance.py

```diff
@@ -126,7 +126,13 @@ def _handle_extra_radiation_types(datetime_or_doy, epoch_year):
     # a better way to do it.
     if isinstance(datetime_or_doy, pd.DatetimeIndex):
         to_doy = tools._pandas_to_doy  # won't be evaluated unless necessary
-        def to_datetimeindex(x): return x                       # noqa: E306
```

Comment:
For the purpose of getting day of year, I think we'll want to convert to UTC for pd.DatetimeIndex, pd.Timestamp, datetime object and np.datetime64. Or, we can do the conversion in the `tools` functions.